src/cli.ts

```diff
@@ -3,29 +3,90 @@ import * as yargs from 'yargs';
 import { Language } from './docgen/transpile/transpile';
 import { Documentation } from './index';
 
+type GenerateOptions = {
+  readme?: boolean;
+  language: Language;
+  submodule?: string;
+  allSubmodules?: boolean;
+  splitBySubmodules?: boolean;
+  format?: 'md' | 'json';
+  output?: string;
+}
+
+async function generateForLanguage(docs: Documentation, options: GenerateOptions) {
+  const { format, output = 'API' } = options;
+  const fileSuffix = format === 'md' ? 'md' : 'json';
+  let submoduleSuffix = fileSuffix;
+
+  // e.g. API.typescript as name
+  if (output.includes('.')) {
+    const languageSeperator = output.split('.')[1];
+    submoduleSuffix = `${languageSeperator}.${fileSuffix}`;
+  }
+
+  if (options.splitBySubmodules) {
+    if (format !== 'md') {
+      throw new Error('split-by-submodule is only supported for markdown');
+    }
+
+    const submodules = await docs.listSubmodules();
+    for (const submodule of submodules) {
+      const content = await docs.toMarkdown({
+        ...options,
+        submodule: submodule.name,
+        allSubmodules: false,
+        header: { title: `\`${submodule.name}\` Submodule`, id: submodule.fqn },
+      });
+
+      fs.writeFileSync(`${submodule.name}.${submoduleSuffix}`, content.render());
+    }
+
+    fs.writeFileSync(`${output ?? 'API'}.${fileSuffix}`, await (await docs.toIndexMarkdown(submoduleSuffix, options)).render());
```

Comment:
Feels a bit weird to use `Sync` FS APIs while you're using `await` there... Can you flip to using the promise versions (either from `fs/promises` or `import { promises as fs } from 'fs';`)?